src/components/BaseButton.vue

```diff
@@ -1,23 +1,33 @@
 <template>
- <button
-    class="inline-flex items-center font-medium rounded-md shadow-sm text-white" type="button"
-   :class="{
-    'cursor-not-allowed': disabled,
-    'px-3 py-2 text-sm': size === 'sm',
-    'px-4 py-2 text-base': size === 'md',
-    'px-6 py-3 text-base': size === 'lg',
+  <button
+    class="inline-flex items-center font-medium text-white rounded-md shadow-sm"
+    type="button"
+    :class="{
+      'cursor-not-allowed': disabled,
+      'px-3 py-2 text-sm': size === 'sm',
+      'px-4 py-2 text-base': size === 'md',
+      'px-6 py-3 text-base': size === 'lg',
 
-    'bg-indigo-600 bg-opacity-70': color === 'primary' && disabled,
-    'text-indigo-600 bg-white border border-indigo-600': color === 'primary' && outline,
-    'bg-indigo-600 active:bg-indigo-800': color === 'primary' && !disabled && !outline,
+      'bg-indigo-600 bg-opacity-70': color === 'primary' && disabled,
+      'text-indigo-600 bg-white border border-indigo-600':
+        color === 'primary' && outline,
+      'bg-indigo-600 active:bg-indigo-800':
+        color === 'primary' && !disabled && !outline,
 
-    'bg-green-600 bg-opacity-70': color === 'secondary' && disabled,
-    'text-green-600 bg-white border border-green-600': color === 'secondary' && outline,
-    'bg-green-600 active:bg-green-800': color === 'secondary' && !disabled && !outline,
+      'bg-green-600 bg-opacity-70': color === 'secondary' && disabled,
+      'text-green-600 bg-white border border-green-600':
+        color === 'secondary' && outline,
+      'bg-green-600 active:bg-green-800':
+        color === 'secondary' && !disabled && !outline,
     }"
     v-bind="$attrs"
-    v-on="$listeners" >
-    <slot>{{label}}</slot>
+    v-on="$listeners"
+    :disabled="disabled"
+  >
+    <i v-if="loading" class="text-lg leading-none el-icon-loading" />
+    <div class="mx-2">
```

Comment:
I would replace this div with a span, and use a v-else on it:
`<i v-if="loading" class="text-lg leading-none el-icon-loading" />
<span v-else class="mx-2">
  <slot>{{ label }}</slot>
</span>`